pkg/chrootarchive/archive.go

```diff
@@ -35,13 +47,25 @@ func Untar(tarArchive io.Reader, dest string, options *archive.TarOptions) error
 		}
 	}
 
-	decompressedArchive, err := archive.DecompressStream(tarArchive)
-	if err != nil {
-		return err
+	var r io.ReadCloser = &noopCloser{Reader: tarArchive}
```

Comment:
It is fine to just use `io.Reader` for `(cmd.Cmd).Stdin`. `cmd.Cmd` keeps its own list of `io.Closer` and does not try to call close anyway like the `http` package does :smile:. If you pass it anything but a file it will just create a Pipe anyway.